src/services/comments.ts

```diff
@@ -80,3 +86,33 @@ export function getCommentsForOption(dbPool: PostgresJsDatabase<typeof db>) {
     }
   };
 }
+async function userCanComment(
+  dbPool: PostgresJsDatabase<typeof db>,
+  userId: string,
+  optionId: string | undefined | null,
+) {
+  if (!optionId) {
+    return false;
+  }
+
+  // check if user has an accepted registration for the option related to the event
+  const res = await dbPool
+    .select()
+    .from(db.questionOptions)
+    .leftJoin(db.forumQuestions, eq(db.forumQuestions.id, db.questionOptions.questionId))
+    .leftJoin(db.cycles, eq(db.cycles.id, db.forumQuestions.cycleId))
+    .leftJoin(db.events, eq(db.events.id, db.cycles.eventId))
+    .leftJoin(db.registrations, eq(db.registrations.eventId, db.events.id))
+    .where(and(eq(db.registrations.userId, userId), eq(db.questionOptions.id, optionId)))
+    .limit(1);
+
+  if (!res.length) {
+    return false;
+  }
+
+  if (res[0]?.registrations?.status !== 'ACCEPTED') {
+    return false;
+  }
+
+  return true;
+}
```

Comment:
I think the user permission checks on the `question_option` level are on a too low level. Why dont we check on a high level like `registrations` directly. In my opinion this gets us the same result with reduced query complexity. I would do it like this:

```
const res = await dbPool
    .selectDistinct({
          id: db.users.id,
        })
    .from(db.registrations)
    .where(and(eq(db.registrations.userId, userId), eq(db.registrations.status, 'ACCEPTED')))
```

 This already satisfies the Berlin usecase. If we want to differentiate between events in the future, we get the eventId as part of the request and check whether the user has `accepted` status by eventId.

Reply:
We go with this solution.